.github/workflows/binary-size-monitor.yml

```diff
@@ -0,0 +1,87 @@
+name: Binary Size Monitor
+
+on:
+  pull_request:
+    branches:
+      - main
+      - "[0-9]+.[0-9]+"
+    types: [opened, synchronize, reopened]
+
+env:
+  # Binary size threshold in MiB
+  SIZE_THRESHOLD_MIB: 5
+
+concurrency:
+  group: ${{ github.workflow }}-${{ github.head_ref || github.run_id }}
```

Comment:
I think the `github.run_id` in the concurrency group essentially means no concurrency group since it's unique per run, not per pr. 

If we want a single concurrency group per pr (so each new commit to cancel the previus job running) then we might need something like `${{ github.event.pull_request.number }}` ([link](https://docs.github.com/en/webhooks/webhook-events-and-payloads#pull_request:~:text=GitHub%20Apps.%22-,number,-integer%20Required)).

Reply:
It's  `github.head_ref || github.run_id` (or), which is a fallback mechanism: https://docs.github.com/en/actions/reference/workflows-and-actions/workflow-syntax#example-using-a-fallback-value

Shouldn't really happen but it doesn't hurt to avoid very hard to debug issues, e.g. if we expand this workflow to different